code/game/objects/machinery/nuclearbomb.dm

```diff
@@ -6,334 +12,244 @@ GLOBAL_LIST_EMPTY(nukes_set_list)
+			if (timing)
+				status = "Set-[safe_text]"
 			else
-				dat += text("\n<B>Status</B>: Auth. S1-[]<BR>\n<B>Timer</B>: []<BR>\n<BR>\nTimer: [] Toggle<BR>\nTime: - - [] + +<BR>\n<BR>\nSafety: [] Toggle<BR>\nAnchor: [] Toggle<BR>\n", (src.safety ? "Safe" : "Engaged"), src.timeleft, (src.timing ? "On" : "Off"), src.timeleft, (src.safety ? "On" : "Off"), (src.anchored ? "Engaged" : "Off"))
-		var/message = "AUTH"
-		if (src.auth)
-			message = text("[]", src.code)
-			if (src.yes_code)
-				message = "*****"
-		dat += text("<HR>\n>[]<BR>\n<A href='?src=\ref[];type=1'>1</A>-<A href='?src=\ref[];type=2'>2</A>-<A href='?src=\ref[];type=3'>3</A><BR>\n<A href='?src=\ref[];type=4'>4</A>-<A href='?src=\ref[];type=5'>5</A>-<A href='?src=\ref[];type=6'>6</A><BR>\n<A href='?src=\ref[];type=7'>7</A>-<A href='?src=\ref[];type=8'>8</A>-<A href='?src=\ref[];type=9'>9</A><BR>\n<A href='?src=\ref[];type=R'>R</A>-<A href='?src=\ref[];type=0'>0</A>-<A href='?src=\ref[];type=E'>E</A><BR>\n", message, src, src, src, src, src, src, src, src, src, src, src, src)
+				status = "Auth. S1-[safe_text]"
+		
+		var/html = {"
+		<b>Nuclear Fission Explosive</b><br />
+		<hr />
+		<b>Status</b>: [status]<br />
+		<b>Timer</b>: [timeleft]<br />
+		<br />
+		Timer: [timing ? "On" : "Off"] [has_auth ? "<a href='?src=[REF(src)];timer=1'>Toggle</a>" : "Toggle"] <br />
+		Time: [has_auth ? "<a href='?src=[REF(src)];time=-10'>--</a> <a href='?src=[REF(src)];time=-1'>-</a> [timeleft] <a href='?src=[REF(src)];time=1'>+</a> <a href='?src=[REF(src)];time=10'>++</a>" : "- - [timeleft] + +"] <br />
+		<br />
+		Safety: [safety ? "On" : "Off"] [has_auth ? "<a href='?src=[REF(src)];safety=1'>Toggle</a>" : "Toggle"] <br />
+		Anchor: [anchored ? "Engaged" : "Off"] [has_auth ? "<a href='?src=[REF(src)];anchor=1'>Toggle</a>" : "Toggle"] <br />
+		<hr />
+		Red Auth. Disk: <a href='?src=[REF(src)];r_auth=1'>[r_auth ? "++++++++++" : "----------"]</a><br />
+		Green Auth. Disk: <a href='?src=[REF(src)];g_auth=1'>[g_auth ? "++++++++++" : "----------"]</a><br />
+		Blue Auth. Disk: <a href='?src=[REF(src)];b_auth=1'>[b_auth ? "++++++++++" : "----------"]</a><br />
+		"}
+
 		var/datum/browser/popup = new(user, "nuclearbomb", "<div align='center'>Nuclear Bomb</div>", 300, 400)
-		popup.set_content(dat)
+		popup.set_content(html)
 		popup.open(FALSE)
 		onclose(user, "nuclearbomb")
-	else if (src.deployable)
-		if(removal_stage < 5)
-			src.anchored = TRUE
-			visible_message("<span class='warning'> With a steely snap, bolts slide out of [src] and anchor it to the flooring!</span>")
+	else if (deployable)
+		if (!do_after(user, 3 SECONDS, TRUE, src, BUSY_ICON_BUILD))
+			return
+		if(removal_stage < NUKE_STAGE_BOLTS_REMOVED)
```

Comment:
If you're gonna use the abomination of `if (` then at least make it consistent across close lines